app-drawer/app-drawer.html

```diff
@@ -552,12 +550,11 @@
 
         if (oldState !== this._drawerState) {
           if (this._drawerState === this._DRAWER_STATE.OPENED) {
+            this._setKeyboardFocusTrap();
             document.addEventListener('keydown', this._boundEscKeydownHandler);
```

Comment:
The current code adds two `keydown` events listeners. I think all the work could be done on a single event listener. Also, there are a few events listeners that are added in the `attached` callback that aren't removed in the `detached` callback.